.github/workflows/build.yml

```diff
@@ -163,6 +163,23 @@ jobs:
       DOCKER_IMAGE: messense/manylinux_2_24-cross:${{ matrix.architecture }}
       TARGET: ${{ matrix.architecture }}-unknown-linux-gnu
     steps:
+
+      - name: Free disk space
+        uses: jlumbroso/free-disk-space@v1.3.1
+        with:
+          tool-cache: false     # no: will remove rust
+          android: true         # 8.7GiB
+          dotnet: true          # 1.6GiB
+          docker-images: true   # 3.2GiB
+          haskell: false        # no: little/no benefit
+          large-packages: false # no: slow
+          swap-storage: false   # no: having swap is useful
+      
+      - name: Set up Python (necessary for PyO3 cross-compilation)
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.8
```

Comment:
- Why does the Rust build need to worry about Python?
- Why is this only needed on linux?
- Could we use a more recent python? Does the build fail downstream if 3.9 or above is specified?